WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,334 @@
+import Foundation
+import Yosemite
+import class WooFoundation.CurrencyFormatter
+
+
+protocol CardPresentPayments {
+
+    /// The adaptor/facade is intended to be created once when we switch stores, and to be a singleton.
+    /// We might as well make a new one when we switch stores though.
+    init(siteID: Int64)
+
+    /// `paymentScreenEventPublisher` provides a stream of events relating to a payment, including their view models,
+    /// for subscribers to display to the user. e.g. onboarding screens, connection progress, payment progress, card reader messages
+    /// We may want separate streams for these depending on the way we consume it
+    var paymentScreenEventPublisher: AnyPublisher<CardPresentPaymentEvent?, Never> { get }
```

Comment:
In this example, which is closely based on the existing UI/UX, `nil` means "don't show any card payment UI", which I think we need to be able to do to remove modals which are finished with (e.g. when cancelling, or when someone taps the last button after a payment.)

That's a good question though, because it's not immediately clear what that would mean if the UI were more ambient, showing the progress in a corner of the screen or similar. In that case, we'd probably always want a state.

Perhaps an `idle` event is required here...